torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3392,34 +3398,54 @@ def error_inputs_embedding(op_info, device, **kwargs):
 
 def error_inputs_multinomial(op_info, device, **kwargs):
     x = torch.empty(1, 2, 3, dtype=torch.double, device=device)
-    yield ErrorInput(SampleInput(x, args=(2,)), error_type=RuntimeError,
+    yield ErrorInput(SampleInput(x, args=(2,)),
                      error_regex="prob_dist must be 1 or 2 dim")
 
     x = torch.empty(1, 2, dtype=torch.long, device=device)
-    yield ErrorInput(SampleInput(x, args=(2,)), error_type=RuntimeError,
+    yield ErrorInput(SampleInput(x, args=(2,)),
                      error_regex="multinomial only supports floating-point dtypes for input")
 
     x = torch.empty(1, 2, dtype=torch.double, device=device)
     y = torch.empty(1, 2, dtype=torch.double, device=device)
-    yield ErrorInput(SampleInput(x, args=(2,), kwargs=dict(out=y)), error_type=RuntimeError,
+    yield ErrorInput(SampleInput(x, args=(2,), kwargs=dict(out=y)),
                      error_regex="multinomial expects Long tensor out")
 
     x = torch.empty(2, dtype=torch.double, device=device)
-    yield ErrorInput(SampleInput(x, args=(0,)), error_type=RuntimeError,
+    yield ErrorInput(SampleInput(x, args=(0,)),
                      error_regex="cannot sample n_sample <= 0 samples")
 
     x = torch.empty(2, dtype=torch.double, device=device)
-    yield ErrorInput(SampleInput(x, args=(-1,)), error_type=RuntimeError,
+    yield ErrorInput(SampleInput(x, args=(-1,)),
                      error_regex="cannot sample n_sample <= 0 samples")
 
     x = torch.empty(2, dtype=torch.double, device=device)
-    yield ErrorInput(SampleInput(x, args=(3, False,)), error_type=RuntimeError,
+    yield ErrorInput(SampleInput(x, args=(3, False,)),
                      error_regex="cannot sample n_sample > prob_dist")
 
     x = torch.empty(16777217, dtype=torch.double, device=device)
-    yield ErrorInput(SampleInput(x, args=(3,)), error_type=RuntimeError,
+    yield ErrorInput(SampleInput(x, args=(3,)),
                      error_regex="number of categories cannot exceed")
 
+    inputs = [[1., -1., 1.], [1., inf, 1.], [1., -inf, 1.], [1., 1., nan]]
+    for input in inputs:
+        yield ErrorInput(SampleInput(torch.tensor(input), kwargs={'num_samples': 2}),
+                         error_regex="probability tensor contains either `inf`, `nan` or element < 0")
+
+    rep_arg = True if device == 'cpu' else False
```

Comment:
I think this has to be `device.type == 'cpu'` otherwise it will always be False.

Also, for CPU you should test with replacement as True and False. Currently for CPU, we only test with True.

Reply:
>  I think this has to be device.type == 'cpu' otherwise it will always be False.

I think the device is of type `str`. 
Some of the similar use-case:
https://github.com/pytorch/pytorch/blob/d0adb5ff264df8e0e057ce1178feb30198c601d2/torch/testing/_internal/common_device_type.py#L852-L860
I tried to use `device.type`, but it throws the following error:
```python
FAILED test/test_ops.py::TestCommonCPU::test_errors_multinomial_cpu - AttributeError: 'str' object has no attribute 'type'
```

Please let me know If I'm missing something :) Thanks!

Reply:
> Also, for CPU you should test with replacement as True and False. Currently for CPU, we only test with True.

Yes, agreed. I missed that. Thanks, Kshitij!! I've updated that.

Reply:
Ah right. Sorry I thought it was a dtype object. Thanks for clarifying.